### PR TITLE
Enable nb api for CAN (TWAI) (embedded-hal 0.2.7)

### PR DESCRIPTION
**Pull request by phansel**
nb::block! macro functions as expected for receive and transmit per example on ESP32-C3 rev 3! 

Thanks @andresv @ivmarkov!

### CODE REVIEW

File: src/can.rs [hunk 1]
```diff
@@ -41,10 +41,13 @@ use crate::gpio::*;
 
 crate::embedded_hal_error!(
     CanError,
-    embedded_hal::can::Error,
-    embedded_hal::can::ErrorKind
+    embedded_hal_0_2::can::Error,
```
Comment (ivmarkov) [inline src/can.rs:44]: I think we need something much more involved than that. Is this even working with e-hal 1.0 alpha-X?

**Review by ivmarkov — CHANGES_REQUESTED** [review]

File: src/can.rs (hunk 1 above)
Comment (phansel) [reply src/can.rs:44]: Oh, that's interesting. It looks like I misunderstood the versioning in Cargo.toml - embedded_hal_0_2 is pre-1.0.0, whereas embedded_hal is 1.0.0-alpha.7 (for the time being). If that's the case, feel free to close this PR; I'll try a different angle.

Comment (phansel) [reply src/can.rs:44]: It does build and work in this form as of embedded-hal 1.0.0-alpha.7.

Comment (ivmarkov) [reply src/can.rs:44]: Yes but the user then needs to depend on both e-hal 0.2 and 1.0. He should be able to only use and depend on one of these.

**Comment by ivmarkov** [comment]
This patch still needs fixing the handling of errors as per [this comment](https://github.com/esp-rs/esp-idf-hal/pull/53#discussion_r835862152) and also needs basic `cargo fmt` applied. Merging as I'll fix these myself post-merge.

**Pull request merged**